### PR TITLE
Include note about Go 1.16 minimum

### PR DESCRIPTION
**Pull request by indrora**
New users who are on older versions of Go may encounter an error with `embed` being required by newer versions of Pulumi. 

Ran into this while I was getting started with the tool -- Debian is shipping 1.15 and Ubuntu LTS ships with 1.13. This won't matter for too long -- Debian will be shipping 1.17 in `testing` soon and the next Ubuntu LTS will likely ship with an up to date version.

### CODE REVIEW

**Comment by leezen** [comment]
Thanks for the suggestion. I'm not sure this change is necessary as the Go language installation page defaults to the latest stable version and includes instructions on determining the version you have installed. I worry that the note will clutter the getting started guides and we already document the supported language versions on the individual language pages.

**Comment by indrora** [comment]
This change brings the Go content more in line with the Node and Python
guidance already in place.

**Comment by leezen** [comment]
@indrora Thanks for the clarification.

**Comment by leezen** [comment]
Let me tag a couple other folks to get their opinions as from my perspective, I'd prefer to keep these notes relatively short and not try to clutter the initial experience. If anything, I'd want to reduce the Python verbiage and keep the details to the language-based pages.

**Comment by leezen** [comment]
Curious on your thoughts @susanev, @cnunciato, and @zchase

**Comment by susanev** [comment]
to make sure we are all on the same page, this is going from ...
before

to
after

i agree with @indrora that this aligns go with the other languages. i hear the point about the amount of content though, i think that we should tackle that problem separately. i also in general worry about sending folks out of the getting started flow in order to get setup.

**Comment by zchase** [comment]
I am fine with this change, but if we want to reduce content I think we should just drop the `code` block and just have the two lines of text. Which is what I think @susanev alluded to above as well?

Seems reasonable that when someone is picking a language during the Getting Started flow that we should let them know the minimum version needed.

File: themes/default/layouts/shortcodes/install-go.html [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+    <div class="note-body">
+        <p>
+            Pulumi requires Go 1.16 or later. If you're using Linux, your distribution may not
+            provide an up to date version of the Go compiler. To check what version of Go you have
+            installed, <code>go version</code> will tell you:
+        </p>
+<pre><code>
+$ go version
+go version go1.17.1 darwin/arm64
+</code></pre>
```
Comment (susanev) [inline themes/default/layouts/shortcodes/install-go.html:19]:
```suggestion
            provide an up to date version of the Go compiler. To check what version of Go you have
            installed, use:<code>go version</code>.
        </p>
```

**Review by susanev — APPROVED** [review]

File: themes/default/layouts/shortcodes/install-go.html (hunk 1 above)
Comment (susanev) [reply themes/default/layouts/shortcodes/install-go.html:19]: @indrora what would you think of removing the code snippet?

Comment (indrora) [reply themes/default/layouts/shortcodes/install-go.html:19]: This looks good

**Pull request merged**